### PR TITLE
Replace assumption on types in 'api' project with explicit type lookup

### PR DESCRIPTION
**Pull request by mbryzek**
Net net this change appears to:
  - solve the issue with losing a namespace on imports in api project
  - cleans up the logic on what it means to be a local type
  - cleans up imports - including only those imports in the oneapi services for types that are used in the service

Also rewrote the downloader to download API specs in batch (vs. one by one) which sped up the REPL loop while debugging. Removed also the warnings on closing asyncHttpClient.

Diff of all artifacts from API build at https://drive.google.com/file/d/1y-YoOeohuJDC4xET1hphmO6h4Lysm5f0/view?usp=sharing

```
ruby diff.rb

Has differences
 - flow-api-event.json
 - flow-api-internal-event.json
 - flow-api-internal.json
 - flow-api-partner.json
 - flow-api.json

Identical
 - api-event-authorization.json
 - api-internal-event-authorization.json
 - api-partner-authorization.json
 - api-partner-proxy.development.config
 - api-partner-proxy.production.config
 - api-partner-proxy.workstation.config
 - flow-api-event-streams.json
 - flow-api-internal-event-streams.json
 - flow-api-misc-event-streams.json
 ```

### CODE REVIEW

**Comment by mbryzek** [comment]
This results in the following diff in generated api.json (excluding imports):

```
<       "type" : "io.flow.apple.pay.v0.models.apple_pay_payment_info",
---
>       "type" : "apple_pay_payment_info",
7636c7126
<       "type" : "io.flow.google.pay.v0.models.payment_data_request",
---
>       "type" : "payment_data_request",
```

Full diff attached

**Comment by mbryzek** [comment]
Screen Shot 2021-09-29 at 9 41 21 AM

**Comment by mbryzek** [comment]
[diff.txt](https://github.com/flowcommerce/api-build/files/7252374/diff.txt)

File: src/main/scala/io/flow/oneapi/Datatype.scala [hunk 1]
```diff
@@ -14,57 +14,63 @@ object TextDatatype {
...
-
 }
 
 /**
   * Parses a text datatype, removing specific namespaces as those
-  * names are expected to be local
+  * names are expected to be local.
   */
-case class TextDatatypeParser() {
+case class TextDatatypeParser(localTypes: Set[String]) {
```
Comment (jlogeart) [inline src/main/scala/io/flow/oneapi/Datatype.scala:23]: Suggest making an `object` with:

```scala
def parse(value: String, localTypes: Set[String])
def toNamespace(value: String): Option[String]
def toTypeLabel(parts: Seq[TextDatatype])
```

There is no need to instantiate a `TextDatatypeParser` for other functions than `parse`. When `parse` is used, the call stack can be replaced by `localTypes: Seq[String]`.

File: src/main/scala/io/flow/oneapi/OneApi.scala [hunk 2]
```diff
@@ -183,11 +174,11 @@ case class OneApi(
     }
   }
 
-  private[this] def normalizeName(parser: TextDatatypeParser, localTypes: Seq[String], service: Service, resource: Resource): Resource = {
+  private[this] def normalizeName(parser: TextDatatypeParser, localTypes: Set[String], service: Service, resource: Resource): Resource = {
```
Comment (jlogeart) [inline src/main/scala/io/flow/oneapi/OneApi.scala:177]: I suggest colocating the creation of the namespaced names and their extraction in `TextDatatypeParser`. That would reduce a bit of the complexity.

**Comment by jlogeart** [comment]
The "namespacization" and extraction logic makes the logic sometimes hard to follow.

I suggest considering introducing a case class such as:

```scala
case class QualifiedType(serviceName: String, `type`: String, name: String) {
  lazy val namespace = serviceName
  ...
  lazy val qualifiedName = s"$serviceName.$`type`.$name"
}
object QualifiedType {
  def apply(service: Service, e: Enum): QualifiedType = 
    QualifiedType(service.name, "enum", e.name)
  ...  
}
```

And using that class as canonical representation or all types, instead of using `String` that is "namespaced" and extracted several times.

**Review by jlogeart — APPROVED** [review]
A couple of comments but overall LGTM.

File: src/main/scala/io/flow/oneapi/Datatype.scala (hunk 1 above)
Comment (mbryzek) [reply src/main/scala/io/flow/oneapi/Datatype.scala:23]: Thanks! I took a pass in this direction - it's more work than I think is worth (intertwined logic between multiple classes)

I think the right end state here is to use the ApiBuilderType class from apibuilder-validation which already encapsulates the namespaces

File: src/main/scala/io/flow/oneapi/OneApi.scala (hunk 2 above)
Comment (mbryzek) [reply src/main/scala/io/flow/oneapi/OneApi.scala:177]: tried this - will take time - suggest same approach as above

**Comment by mbryzek** [comment]
Agreed on QualifiedType! My thought here was in future to move this all to https://github.com/apicollective/apibuilder-validation/blob/main/src/main/scala/io/apibuilder/validation/AnyType.scala#L47 which is already modeled
